Siri/IntentHandler.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  IntentHandler.swift
+//  Siri
+//
+//  Created by Yana Sang on 10/17/18.
+//  Copyright © 2018 cuappdev. All rights reserved.
+//
+
+import Intents
+
+class IntentHandler: INExtension {
+
+    override func handler(for intent: INIntent) -> Any {
+        if #available(iOS 12.0, *) {
+            switch intent {
+            case is GetRoutesIntent:
+                return GetRoutesIntentHandler()
+            default:
+                return self
+            }
+        } else {
+            return self
+        }
+    }
+}
+
+@available(iOS 12.0, *)
+class GetRoutesIntentHandler: INExtension, GetRoutesIntentHandling {
+    func handle(intent: GetRoutesIntent, completion: @escaping (GetRoutesIntentResponse) -> Void) {
+        if let _ = intent.latitude, let _ = intent.longitude, let _ = intent.searchTo {
```

Comment:
nitpick: If you don't use the variable, it's preferred to directly check for nil (`if intent.latitude != nil`) You can also use 

```
guard intent.latitude != nil, intent.longitude != nil ... else { 
    completion(.failure)
}
```